presto-product-tests/src/main/java/io/prestosql/tests/hive/TestHiveTransactionalTable.java

```diff
@@ -111,15 +126,21 @@ public void testReadInsertOnly(boolean isPartitioned, BucketingType bucketingTyp
             assertThat(query(selectFromOnePartitionsSql)).containsExactly(row(1), row(2));
 
             // test minor compacted data read
-            onHive().executeQuery("ALTER TABLE " + tableName + " " + hivePartitionString + " COMPACT 'MINOR' AND WAIT");
+            compactTableAndWait(MINOR, tableName, hivePartitionString, Duration.valueOf("5m"));
             assertThat(query(selectFromOnePartitionsSql)).containsExactly(row(1), row(2));
 
             onHive().executeQuery("INSERT OVERWRITE TABLE " + tableName + hivePartitionString + " SELECT 3");
             assertThat(query(selectFromOnePartitionsSql)).containsOnly(row(3));
 
+            if (getHiveVersionMajor() < 4) {
```

Comment:
nit: I would rather wrap the assertion with `getHiveVersionMajor() >= 0`. Otherwise one can not notice the `return` in the middle of the test and add assertion at the end. Then it will not be executed for Hive <= 4.0.

Reply:
Makes sense :)